src/functionalTest/groovy/com/autonomousapps/jvm/IncludedBuildSpec.groovy

```diff
@@ -19,15 +20,34 @@ final class IncludedBuildSpec extends AbstractJvmSpec {
     gradleProject = project.gradleProject
 
     when: 'build does not fail'
-    build(gradleVersion, gradleProject.rootDir, 'buildHealth')
+    build(gradleVersion, gradleProject.rootDir, ':buildHealth')
+    build(gradleVersion, gradleProject.rootDir, ':second-build:buildHealth')
 
-    then: 'and there is no advice'
-    assertThat(project.actualBuildHealth()).containsExactlyElementsIn(project.expectedBuildHealth)
+    then: 'and there is the expected advice'
+    assertThat(project.actualBuildHealth()).containsExactlyElementsIn(project.expectedBuildHealth('second-build'))
+    assertThat(project.actualBuildHealthOfSecondBuild()).containsExactlyElementsIn(project.expectedBuildHealthOfIncludedBuild(':'))
 
     where: 'This new feature only works for Gradle 7.3+'
     gradleVersion << INCLUDED_BUILD_SUPPORT_GRADLE_VERSIONS
   }
 
+  def "result of analysis does not change if root build of included build tree changes (only root projects)"() {
```

Comment:
by `(only root projects)`, do you mean that we're only analyzing the root projects? that would be in accord with invoking the task as `:buildHealth` instead of `buildHealth`.

Reply:
No. It's about the projects not having subprojects that reference each other. I changed the test name.